python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -328,8 +333,85 @@ def get_trainer_program(self):
         # FIXME(typhoonzero): Also ops like clip_gradient, lrn_decay?
         delete_ops(self.origin_program.global_block(), self.optimize_ops)
         self.origin_program.__str__()
+
         return self.origin_program
 
+    def _get_trainer_startup_program(self, program=None):
+        """
+        Get transpiled trainer side startup program.
+
+        Returns:
+            Program: trainer side startup program.
+        """
+        if program is None:
+            program = self.startup_program
+
+        orig_s_prog = program
+
+        # add concat ops to origin parameters in startup program to
+        # let the origin parameters initialized by the spilited parameters
+        send_vars = []
+
+        for orig_varname, splited_vars in self.grad_var_mapping.items():
+            for _, var in enumerate(splited_vars):
+                send_vars.append(var)
+
+        recv_vars = []
+        for _, var in enumerate(send_vars):
+            recv_vars.append(self.grad_param_mapping[var])
+        ps_dispatcher = self.split_method(self.pserver_endpoints)
+        ps_dispatcher.reset()
+        eplist = ps_dispatcher.dispatch(recv_vars)
```

Comment:
Better to call `dispatch` for once, so that `recv_vars` for each server is deterministic in every call.

Reply:
Done.